cluster-autoscaler/simulator/clustersnapshot/predicate/predicate_snapshot_dra_benchmark_test.go

```diff
@@ -0,0 +1,384 @@
+		"10x32x50/MixedClaims/ForkRevert":            {sharedClaimPods: 32, ownedClaimPods: 50, nodesCount: 10, forks: 1, reverts: 1},
+		"10x32x50/MixedClaims/ForkCommit":            {sharedClaimPods: 32, ownedClaimPods: 50, nodesCount: 10, forks: 1, commits: 1},
+		"10x32x50/MixedClaims/ForkForkCommitRevert":  {sharedClaimPods: 32, ownedClaimPods: 50, nodesCount: 10, forks: 2, reverts: 1, commits: 1},
+		"10x32x50/MixedClaims/Fork":                  {sharedClaimPods: 32, ownedClaimPods: 50, nodesCount: 10, forks: 1},
+		"10x32x50/MixedClaims/Fork5Revert5":          {sharedClaimPods: 32, ownedClaimPods: 50, nodesCount: 10, forks: 5, reverts: 5},
+		"10x1x1/MixedClaims/ForkRevert":              {sharedClaimPods: 1, ownedClaimPods: 1, nodesCount: 10, forks: 1, reverts: 1},
+		"10x1x1/MixedClaims/ForkCommit":              {sharedClaimPods: 1, ownedClaimPods: 1, nodesCount: 10, forks: 1, commits: 1},
+		"10x1x1/MixedClaims/ForkForkCommitRevert":    {sharedClaimPods: 1, ownedClaimPods: 1, nodesCount: 10, forks: 2, reverts: 1, commits: 1},
+		"10x1x1/MixedClaims/Fork":                    {sharedClaimPods: 1, ownedClaimPods: 1, nodesCount: 10, forks: 1},
+		"10x1x1/MixedClaims/Fork5Revert5":            {sharedClaimPods: 1, ownedClaimPods: 1, nodesCount: 10, forks: 5, reverts: 5},
+	}
+
+	devicesClasses := map[string]*resourceapi.DeviceClass{
+		deviceClassName: {ObjectMeta: metav1.ObjectMeta{Name: deviceClassName, UID: "defaultClassUid"}},
+	}
+
+	nodeInfos := make([]*framework.NodeInfo, maxNodesCount)
+	sharedClaims := make([]*resourceapi.ResourceClaim, maxNodesCount)
+	ownedClaims := make([][]*resourceapi.ResourceClaim, maxNodesCount)
+	owningPods := make([][]*apiv1.Pod, maxNodesCount)
+	for nodeIndex := 0; nodeIndex < maxNodesCount; nodeIndex++ {
+		nodeName := fmt.Sprintf("node-%d", nodeIndex)
+		node := BuildTestNode(nodeName, 10000, 10000)
+		nodeSlice := createTestResourceSlice(node.Name, devicesPerSlice, 1, driverName, resourceapi.BasicDevice{})
+		nodeInfo := framework.NewNodeInfo(node, []*resourceapi.ResourceSlice{nodeSlice})
+
+		sharedClaim := createTestResourceClaim(devicesPerSlice, 1, driverName, deviceClassName)
+		sharedClaim, satisfied := allocateResourceSlicesForClaim(sharedClaim, nodeName, nodeSlice)
+		if !satisfied {
+			b.Errorf("Error during setup, claim allocation cannot be satistied")
+		}
+
+		claimsOnNode := make([]*resourceapi.ResourceClaim, maxPodsCount)
+		podsOnNode := make([]*apiv1.Pod, maxPodsCount)
+		for podIndex := 0; podIndex < maxPodsCount; podIndex++ {
+			podName := fmt.Sprintf("pod-%d-%d", nodeIndex, podIndex)
+			ownedClaim := createTestResourceClaim(1, 1, driverName, deviceClassName)
+			pod := BuildTestPod(
+				podName,
+				1,
+				1,
+				WithResourceClaim(ownedClaim.Name, ownedClaim.Name, ""),
+			)
+
+			ownedClaim = drautils.TestClaimWithPodOwnership(pod, ownedClaim)
+			ownedClaim, satisfied := allocateResourceSlicesForClaim(ownedClaim, nodeName, nodeSlice)
```

Comment:
I don't think we should be benchmarking on pre-allocated claims. In a typical scenario, at least most of the pod owned claims should be unallocated and require the DRA scheduler plugin to compute the allocation.